kotlin/workflow-core/src/main/java/com/squareup/workflow/Workflow.kt

```diff
@@ -73,4 +75,63 @@ interface Workflow<in InputT : Any, out OutputT : Any, out RenderingT : Any> {
    * the common API required for [WorkflowContext.compose] to do its work.
    */
   fun asStatefulWorkflow(): StatefulWorkflow<InputT, *, OutputT, RenderingT>
+
+  companion object {
+
+    /**
+     * A convenience function to implement [stateless] by just passing the
+     * [compose][stateless.compose] function as a lambda.
+     *
+     * Note that while a stateless workflow doesn't have any _internal_ state of its own, it may use
+     * [input][InputT] received from its parent, and it may compose child workflows that do have their own
+     * internal state.
+     */
+    fun <InputT : Any, OutputT : Any, RenderingT : Any> stateless(
```

Comment:
How would you feel about keeping `Workflow`'s companion object empty and put these functions back in `StatelessWorkflow.kt` as extension methods on it?

Reply:
I'm really liking `Workflow.stateless`.

Reply:
Even find myself wondering if we can make that abstract class internal.

Reply:
I like that too, I want to keep it I just want to make it an extension function to keep this file more focused.

Reply:
And we could make it internal, but it might be more convenient to just subclass it if you want to take constructor params from Dagger, e.g.